invenio_rdm_records/records/jsonschemas/records/record-v1.0.0.json

```diff
@@ -1,402 +1,714 @@
+          "description": "Checksum of the file.",
+          "type": "string"
+        },
+        "size": {
+          "description": "Size of the file in bytes.",
+          "type": "number"
+        },
+        "key": {
+          "description": "Key (filename) of the file.",
+          "type": "string"
+        },
+        "ext": {
+          "description": "File extension.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description for file.",
+          "type": "string"
+        },
+        "order": {
+          "description": "Value to sort according to.",
+          "type": "string"
+        },
+        "default_preview": {
+          "description": "Set to true to use as the default previewed file.",
+          "type": "boolean"
+        },
+        "bucket": {
+          "description": "TODO - Bucket id - what about third party storage systems?",
+          "type": "string"
+        },
+        "file_id": {
+          "description": "TODO - Identifier of the file. - same as for bucket",
+          "type": "string"
         }
-      },
-      "dates": {
-        "description": "Date or date interval.",
-        "minItems": 0,
-        "type": "array",
-        "items": {
-          "type": "object",
-          "additionalProperties": false,
-          "properties": {
-            "description": {
-              "description": "Description of the date or date interval e.g. 'Accepted' or 'Available' (CV).",
-              "type": "string"
-            },
-            "end": {
-              "description": "End date.",
-              "type": "string",
-              "format": "date-time"
-            },
-            "start": {
-              "description": "Start date.",
-              "type": "string",
-              "format": "date-time"
-            },
-            "type": {
-              "description": "Type of the date interval."
-            }
-          },
-          "required": ["type"]
+      }
+    },
+
+    "agent-user": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "user": {
+          "type": "integer"
         }
-      },
-      "descriptions": {
-        "description": "Description for record.",
-        "type": "array",
-        "uniqueItems": true,
-        "items": {
+      }
+    },
+
+    "agent": {
+      "description": "An agent.",
+      "oneOf": [
+        {"$ref": "#/definitions/agent-user"}
+      ]
+    }
+  },
+
+
+  "properties": {
+    "$schema": {
+      "description": "JSONSchema declaration.",
+      "type": "string"
+    },
+    "id": {
+      "description": "Persistent record identifier (alphanumeric).",
+      "type": "string"
+    },
+    "pid": {"$ref": "#/definitions/internal-pid"},
+
+    "conceptid": {
+      "description": "Persistent concept record identifier (alphanumeric).",
+      "type": "string"
+    },
+    "conceptpid": {"$ref": "#/definitions/internal-pid"},
+
+    "pids": {
```

Comment:
I assume is not an array because we want the type of pid as key. Would this generate a possible mapping explosion in ES? Or will we have a not 1-to-1 mapping?

Reply:
Maybe need transformation. But not expecting many keys.